helpers/setup_json_generator.py

```diff
@@ -0,0 +1,297 @@
+                    osd_dict = osds_on_disks
+                elif scaling == 'GLOBAL':
+                    if linked_backend not in self._backends:
+                        raise ValueError("Provided backend {} not in known backends".format(linked_backend))
+                    if linked_preset not in self._presets:
+                        raise ValueError("Provided preset {} not in known presets".format(linked_preset))
+                    osd_dict = {linked_backend: linked_preset}
+
+                else:
+                    print ValueError('invalid scaling ({0}) passed'.format(scaling))
+            self.json['setup']['backends'][i]['osds'].update(osd_dict)
+
+    def add_vpool(self, storagerouter_ip, vpool_name, backend_name, preset, storage_ip):
+        if backend_name not in self._backends:
+            raise ValueError("Provided backend {} not in known backends".format(backend_name))
+        if preset not in self._presets:
+            raise ValueError('Provided preset not in known presets'.format(preset))
+
+        vpool_dict = {'backend_name': backend_name,
+                      'preset': preset,
+                      'storage_ip': storage_ip,
+                      'proxies':1,
+                      'fragment_cache': {
+                          'strategy': {'cache_on_read': False, 'cache_on_write': False},
+                          'location': "disk"
+                      },
+                      'block_cache': {
+                          'strategy': {'cache_on_read': False, 'cache_on_write': False},
+                          'location': "disk"
+                      },
+                      'storagedriver':
+                          {
+                              "sco_size": 4,
+                              "cluster_size": 4,
+                              "volume_write_buffer": 512,
+                              "strategy": "none",
+                              "global_write_buffer": 20,
+                              "global_read_buffer": 0,
+                              "deduplication": "non_dedupe",
+                              "dtl_transport": "tcp",
+                              "dtl_mode": "sync"
+                          }
+                      }
+        self.json['setup']['storagerouters'][storagerouter_ip]['vpools'][vpool_name] = vpool_dict
+
+    def _change_cache(self, storagerouter_ip, vpool, block_cache=True, fragment_cache=True, on_read=True, on_write=True):
+        try:
+            vpool = self.json['setup']['storagerouters'][storagerouter_ip]['vpools'][vpool]
+            if block_cache is True:
+                vpool['block_cache']['strategy']['cache_on_read'] = on_read
+                vpool['block_cache']['strategy']['cache_on_write'] = on_write
+            if fragment_cache is True:
+                vpool['fragment_cache']['strategy']['cache_on_read'] = on_read
+                vpool['fragment_cache']['strategy']['cache_on_write'] = on_write
+
+        except KeyError as e:
```

Comment:
Kinda useless error message
The message of a `KeyError` is always the key which it couldn't find
So re-raising this with a `ValueError` is unnecessary, just let it raise the `KeyError` or show a more relevant error message